src/TokenHandler.mo

```diff
@@ -307,6 +316,18 @@ module TokenHandler {
     /// query the fee
     public func getFee() : Nat = fee_;
 
+    /// load fee from ICRC1 ledger. Returns actual fee if successful
+    public func updateFee() : async* ?Nat = async* try {
+      let newFee = await icrc1Ledger.icrc1_fee();
+      if (fee_ != newFee) {
+        journal.push((Time.now(), ownPrincipal, #feeUpdated({ old = fee_; new = newFee })));
+        fee_ := newFee;
+      };
+      ?newFee;
+    } catch (err) {
+      null;
+    };
+
```

Comment:
For example `notify()` passes through asynchronous errors (by re-throwing them). For consistency we could do the same here. We can either do `throw err` in the catch branch or we can have no try-catch at all. Without a surrounding try-catch an asynchronous error will be "uncaught" in which case it will propagate upwards, which will be the same as re-throwing.

Alternatively, to unify the handling we can also make notify() return null in its catch branch.

For debugging purposes it may be beneficial to pass the asynchronous error through instead of converting it to null because it contains an error messge.

Reply:
Done